src/main/java/org/sonarsource/scanner/jenkins/pipeline/WaitForQualityGateStep.java

```diff
@@ -127,6 +142,7 @@ private void processStepParameters() throws IOException, InterruptedException {
       for (SonarAnalysisAction a : actions) {
         if (a.getCeTaskId() != null) {
           step.setTaskId(a.getCeTaskId());
+          step.setServerUrl(a.getServerUrl());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Annotate the parameter with @javax.annotation.Nullable in method 'setServerUrl' declaration, or make sure that null can not be passed as argument. ([squid:S4449](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS4449))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.jenkins-ci.plugins%3Asonar&pullRequest=90&issues=AWIwh_zS_wH-bYHl3dHw&open=AWIwh_zS_wH-bYHl3dHw)

[ISSUE_KEY]: AWIwh_zS_wH-bYHl3dHw